pkg/experiment/ingester/segment.go

```diff
@@ -284,38 +280,32 @@ func (s *segment) flushBlock(heads []serviceHead) (string, *metastorev1.BlockMet
 
 	meta.Size = uint64(w.offset)
 	s.debuginfo.flushBlockDuration = time.Since(t1)
-	return blockPath, meta, nil
+	return blockFile.Bytes(), meta, nil
 }
 
-func concatSegmentHead(sh *shard, e serviceHead, w *writerOffset) (*metastorev1.Dataset, error) {
+func concatSegmentHead(e flushedServiceHead, w *writerOffset) (*metastorev1.Dataset, error) {
 	tenantServiceOffset := w.offset
-	b := e.head.Meta()
-	ptypes := e.head.MustProfileTypeNames()
 
-	profiles, index, symbols := getFilesForSegment(e.head, b)
+	ptypes := e.head.Meta.ProfileTypeNames
 
-	offsets := make([]uint64, 3)
-	var err error
-	offsets[0], err = concatFile(w, e.head, profiles, sh.concatBuf)
-	if err != nil {
-		return nil, err
-	}
-	offsets[1], err = concatFile(w, e.head, index, sh.concatBuf)
-	if err != nil {
-		return nil, err
-	}
-	offsets[2], err = concatFile(w, e.head, symbols, sh.concatBuf)
-	if err != nil {
-		return nil, err
-	}
+	offsets := []uint64{0, 0, 0}
+
+	offsets[0] = uint64(w.offset)
+	_, _ = w.Write(e.head.Profiles)
+
+	offsets[1] = uint64(w.offset)
+	_, _ = w.Write(e.head.Index)
+
+	offsets[2] = uint64(w.offset)
+	_, _ = w.Write(e.head.Symbols)
 
 	tenantServiceSize := w.offset - tenantServiceOffset
 
 	svc := &metastorev1.Dataset{
 		TenantId: e.key.tenant,
 		Name:     e.key.service,
-		MinTime:  int64(b.MinTime),
-		MaxTime:  int64(b.MaxTime),
+		MinTime:  e.head.Meta.MinTimeNanos / 1e6,
+		MaxTime:  e.head.Meta.MaxTimeNanos / 1e6,
```

Comment:
Hm, I can't remember what was the unit – note that this may require change in the metastore. I'd say that we should use nanoseconds everywhere but HTTP API

Reply:
created an issue to not forget https://github.com/orgs/grafana/projects/425/views/9?pane=issue&itemId=76384372